### ISSUE TITLE
"database error: stack depth limit exceeded" when getting territoire infos with lots of oracle URLs

### ISSUE THREAD

**Issue by DavidBruant** [MyWebIntelligence/MyWebIntelligence#123]
Likely a query size problem. Likely related to the number of "OR"s (which are nested and would make the query parser break). See discussion at:
http://www.postgresql.org/message-id/BAY176-W382A9DE827EBC8E602B7BBC5860@phx.gbl
http://www.postgresql.org/message-id/2754.1393260006@sss.pgh.pa.us
http://www.postgresql.org/message-id/BAY176-W37F28B75E21E95BCF9FBF9C5860@phx.gbl
http://www.postgresql.org/message-id/5769.1393268306@sss.pgh.pa.us

**Comment by DavidBruant** [comment]
http://www.postgresql.org/message-id/2754.1393260006@sss.pgh.pa.us is particularly interesting:

> Even without the stack depth issue, that would perform pretty horridly for
so many pairs.

> Do you know that the pairs are all distinct, so that you don't really need
the duplicate-elimination behavior of IN?  If so, you could recast this
like so:

> ```sql
select count(*) from
  gui_die_summary,
  (values (25,5),(41,13),(25,7),(28,3),(25,8),(34,7),(26,6),(21,10)) v(vx,vy)
where (x_coord, y_coord) = (vx, vy);
```

> which should work better for large numbers of pairs.


Should probably try something like that.

**Comment by DavidBruant** [comment]
How things are stored will be reworked soon enough to add a proper "page index". Expressions will have an associated territoire. This should simplify how the graph is fetched from the database.

**Comment by DavidBruant** [comment]
Fixed at #130. Worked like a charm.
